components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/services/UserIdentityManagementAdminService.java

```diff
@@ -610,18 +636,59 @@ private String getUserStoreDomainName(String userName) {
         return userNameWithoutDomain;
     }
 
-    private void validateSecurityQuestionDuplicate(UserChallengesDTO[] challengesDTOs) throws IdentityMgtServiceException {
+    private void validateSecurityQuestionDuplicate(UserChallengesDTO[] challengesDTOs)
+            throws IdentityMgtServiceException {
 
         Set<String> tmpMap = new HashSet<String>();
-        for(int i = 0; i < challengesDTOs.length ; i++) {
+        for (int i = 0; i < challengesDTOs.length; i++) {
             UserChallengesDTO userChallengesDTO = challengesDTOs[i];
-            if(tmpMap.contains(userChallengesDTO.getId())){
-                String errMsg = "Error while validating user challenges, because these can't be more than one security challenges for one claim uri" ;
+            if (tmpMap.contains(userChallengesDTO.getId())) {
+                String errMsg = "Error while validating user challenges, because these can't be " +
+                        "more than one security challenges for one claim uri";
                 log.error(errMsg);
                 throw new IdentityMgtServiceException(errMsg);
             }
             tmpMap.add(userChallengesDTO.getId());
         }
     }
 
+    /**
+     * Check if the request updates super admin properties.
+     *
+     * @param userName Username
+     * @return boolean
+     */
+    private boolean isAttemptToUpdateSuperAdmin(String userName) throws org.wso2.carbon.user.core.UserStoreException {
+
+        String loggedInUserName = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (loggedInUserName != null) {
+            loggedInUserName = addPrimaryDomainIfNotExists(loggedInUserName);
+        }
+        UserRealm realm = (UserRealm) CarbonContext.getThreadLocalCarbonContext().getUserRealm();
+        RealmConfiguration realmConfig = realm.getRealmConfiguration();
+        String adminUser = addPrimaryDomainIfNotExists(realmConfig.getAdminUserName());
+        if (realmConfig.getAdminUserName().equalsIgnoreCase(userName) &&
+                !adminUser.equalsIgnoreCase(loggedInUserName)) {
```

Comment:
What you are doing is not reflected in the method name.

Reply:
Renamed method.